test/e2e/storage/drivers/in_tree.go

```diff
@@ -260,11 +258,16 @@ func (g *glusterFSDriver) GetDriverInfo() *testsuites.DriverInfo {
 	return &g.driverInfo
 }
 
-func (g *glusterFSDriver) SkipUnsupportedTest(pattern testpatterns.TestPattern) {
-	framework.SkipUnlessNodeOSDistroIs("gci", "ubuntu", "custom")
+func (g *glusterFSDriver) IsTestSupported(pattern testpatterns.TestPattern) bool {
+	if !framework.NodeOSDistroIs("gci", "ubuntu", "custom") {
+		return false
+	}
 	if pattern.FsType == "xfs" {
-		framework.SkipUnlessNodeOSDistroIs("ubuntu", "custom")
+		if !framework.NodeOSDistroIs("ubuntu", "custom") {
+			return false
+		}
 	}
+	return true
```

Comment:
nit:

Something like this is arguably easier to read:

```golang
if framework.NodeOSDistroIs("gci", "ubuntu", "custom") {
    if pattern.FsType != "xfs" || framework.NodeOSDistroIs("ubuntu", "custom") {
        return true
    }
}
return false
```
To improve it further, I would personally use this (but up to you):
```golang
return framework.NodeOSDistroIs("gci", "ubuntu", "custom") &&
       (pattern.FsType != "xfs" ||
        framework.NodeOSDistroIs("ubuntu", "custom"))
```

The double negatives in your code made it a bit hard for me to parse

Reply:
I also like the latter better, but again didn't want to change tooo much to keep the review simpler: when a `ginkgo.Skip` gets replaced by a `return true` or now `return true`, then it is obvious that the overall logic is unchanged.

I'll change it.

Reply:
Done.